### ISSUE TITLE
Debian Jessie: apt-get update fails, no network connectivity in LXC box?

### ISSUE THREAD

**Issue by djvdorp** [fgrehm/vagrant-lxc-base-boxes#35]
I am trying to build a Debian Jessie x86_64 Vagrant LXC base box.
Since there are [some issues building this on Debian itself](https://github.com/fgrehm/vagrant-lxc/wiki/Usage-on-debian-hosts), I have decided to spin up a fresh Amazon EC2 instance (HVM, Ubuntu 14.04 LTS) for this task. LXC version that is Installed: 1.0.7-0ubuntu0.1.

Right after installing, this is what I have done:
```
$ sudo su
# apt-get update
# apt-get install lxc htop nmon git screen make
# git clone https://github.com/fgrehm/vagrant-lxc-base-boxes.git
# cd vagrant-lxc-base-boxes
# make jessie
```

This fails with the following error, which is also found in [this other issue](https://github.com/fgrehm/vagrant-lxc-base-boxes/issues/10), but I do not feel like bumping that up since it is over a year old and I do not try to create the Vagrant LXC base box on Debian but on Ubuntu on purpose.
```
==> [debian-jessie] Building box to 'output/2015-06-09/vagrant-lxc-jessie-amd64.box'...
    [debian-jessie] Creating container...
    [debian-jessie] Container created!
    [debian-jessie] Adding ipv6 allhosts entry to container's /etc/hosts
    [debian-jessie] Running [/usr/sbin/update-rc.d -f checkroot-bootclean.sh remove] inside 'vagrant-base-jessie-amd64' container...
    [debian-jessie] Running [/usr/sbin/update-rc.d -f mountall-bootclean.sh remove] inside 'vagrant-base-jessie-amd64' container...
    [debian-jessie] Running [/usr/sbin/update-rc.d -f mountnfs-bootclean.sh remove] inside 'vagrant-base-jessie-amd64' container...
    [debian-jessie] Running [/usr/sbin/locale-gen en_US.UTF-8] inside 'vagrant-base-jessie-amd64' container...
    [debian-jessie] Running [update-locale LANG=en_US.UTF-8] inside 'vagrant-base-jessie-amd64' container...
==> [debian-jessie] Installing extra packages and upgrading
    [debian-jessie] Sleeping for 5 seconds...
    [debian-jessie] Running [apt-get update] inside 'vagrant-base-jessie-amd64' container...
    [debian-jessie] Running [apt-get install vim nfs-common curl wget man-db bash-completion python-software-properties ca-certificates sudo -y --force-yes] inside 'vagrant-base-jessie-amd64' container...
make: *** [jessie] Error 100
```

If I do ```lxc-attach -n vagrant-base-jessie-amd64``` I can get the following network debug information:
```
root@vagrant-base-jessie-amd64:/# ifconfig
eth0      Link encap:Ethernet  HWaddr 00:16:3e:84:e6:b5  
          inet6 addr: fe80::216:3eff:fe84:e6b5/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:8 errors:0 dropped:0 overruns:0 frame:0
          TX packets:8 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:648 (648.0 B)  TX bytes:648 (648.0 B)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

root@vagrant-base-jessie-amd64:/# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
8: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 00:16:3e:84:e6:b5 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::216:3eff:fe84:e6b5/64 scope link 
       valid_lft forever preferred_lft forever
root@vagrant-base-jessie-amd64:/# ip r
root@vagrant-base-jessie-amd64:/# cat /etc/network/interfaces
auto lo
iface lo inet loopback

auto eth0
iface eth0 inet dhcp
root@vagrant-base-jessie-amd64:/# cat /etc/resolv.conf 
# Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
#     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
nameserver 10.0.3.1
```

**Comment by djvdorp** [comment]
This might be helpful to get some more in-depth info to compare with the ```make wheezy``` task which works successfully and has network connectivity:
```
# make wheezy
```

Output:
```
==> [debian-wheezy] Building box to 'output/2015-06-09/vagrant-lxc-wheezy-amd64.box'...
    [debian-wheezy] Creating container...
    [debian-wheezy] Container created!
    [debian-wheezy] Adding ipv6 allhosts entry to container's /etc/hosts
    [debian-wheezy] Running [/usr/sbin/update-rc.d -f checkroot-bootclean.sh remove] inside 'vagrant-base-wheezy-amd64' container...
    [debian-wheezy] Running [/usr/sbin/update-rc.d -f mountall-bootclean.sh remove] inside 'vagrant-base-wheezy-amd64' container...
    [debian-wheezy] Running [/usr/sbin/update-rc.d -f mountnfs-bootclean.sh remove] inside 'vagrant-base-wheezy-amd64' container...
    [debian-wheezy] Running [/usr/sbin/locale-gen en_US.UTF-8] inside 'vagrant-base-wheezy-amd64' container...
    [debian-wheezy] Running [update-locale LANG=en_US.UTF-8] inside 'vagrant-base-wheezy-amd64' container...
==> [debian-wheezy] Installing extra packages and upgrading
    [debian-wheezy] Sleeping for 5 seconds...
    [debian-wheezy] Running [apt-get update] inside 'vagrant-base-wheezy-amd64' container...
    [debian-wheezy] Running [apt-get install vim nfs-common curl wget man-db bash-completion python-software-properties ca-certificates sudo -y --force-yes] inside 'vagrant-base-wheezy-amd64' container...
    [debian-wheezy] Running [apt-get upgrade -y --force-yes] inside 'vagrant-base-wheezy-amd64' container...
    [debian-wheezy] Skipping Chef installation
    [debian-wheezy] Skipping Puppet installation
    [debian-wheezy] Skipping Salt installation
    [debian-wheezy] Skipping Babushka installation
==> [debian-wheezy] Preparing vagrant user...
    [debian-wheezy] SSH credentials configured for the vagrant user.
    [debian-wheezy] Sudoers file created.
==> [debian-wheezy] Cleaning up 'vagrant-base-wheezy-amd64'...
    [debian-wheezy] Removing temporary files...
    [debian-wheezy] cleaning up dhcp leases
    [debian-wheezy] Removing downloaded packages...
    [debian-wheezy] Running [apt-get clean] inside 'vagrant-base-wheezy-amd64' container...
==> [debian-wheezy] Packaging 'vagrant-base-wheezy-amd64' to 'output/2015-06-09/vagrant-lxc-wheezy-amd64.box'...
    [debian-wheezy] Removing previous rootfs tarball
    [debian-wheezy] Compressing container's rootfs
    [debian-wheezy] Preparing box package contents
    [debian-wheezy] Packaging box
==> [debian-wheezy] Finished building 'output/2015-06-09/vagrant-lxc-wheezy-amd64.box'!
    [debian-wheezy] Run `sudo lxc-destroy -n vagrant-base-wheezy-amd64` or `make clean` to remove the container that was created along the way
```

I can also get the following network debug information there:
```
# lxc-start -n vagrant-base-wheezy-amd64 -d
# lxc-attach -n vagrant-base-wheezy-amd64
```

Output:
```
root@vagrant-base-wheezy-amd64:/# ifconfig
eth0      Link encap:Ethernet  HWaddr 00:16:3e:f3:5d:c2  
          inet addr:10.0.3.26  Bcast:10.0.3.255  Mask:255.255.255.0
          inet6 addr: fe80::216:3eff:fef3:5dc2/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:11 errors:0 dropped:0 overruns:0 frame:0
          TX packets:12 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1393 (1.3 KiB)  TX bytes:1524 (1.4 KiB)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

root@vagrant-base-wheezy-amd64:/# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
12: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 00:16:3e:f3:5d:c2 brd ff:ff:ff:ff:ff:ff
    inet 10.0.3.26/24 brd 10.0.3.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::216:3eff:fef3:5dc2/64 scope link 
       valid_lft forever preferred_lft forever
root@vagrant-base-wheezy-amd64:/# ip r
default via 10.0.3.1 dev eth0 
10.0.3.0/24 dev eth0  proto kernel  scope link  src 10.0.3.26 
root@vagrant-base-wheezy-amd64:/# cat /etc/network/interfaces
auto lo
iface lo inet loopback

auto eth0
iface eth0 inet dhcp
root@vagrant-base-wheezy-amd64:/# cat /etc/resolv.conf
nameserver 10.0.3.1
```

**Comment by djvdorp** [comment]
Fixed, the error seems to be in LXC version 1.0.7-0ubuntu0.1 from Ubuntu's own repo.

Doing this solved it: ```add-apt-repository ppa:ubuntu-lxc/stable && apt-get update && apt-get install lxc```

PPA ubuntu-lxc/stable installs version 1.1.2-0ubuntu3 instead which gives ipv4 and finishes building successfully.
